crates/configuration/src/parachain.rs

```diff
@@ -0,0 +1,147 @@
+use crate::shared::{
+    node::NodeConfig,
+    types::{Command, MultiAddress, Path},
+};
+
+/// A parachain configuration, composed of collators and fine-grained configuration options.
+#[derive(Clone)]
+pub struct ParachainConfig {
+    // Parachain ID to use.
+    id: u16,
+
+    /// Chain to use (use None if you are running adder-collator or undying-collator).
+    chain: Option<String>,
+
+    /// Wether to add the parachain to the genesis (chain specification) file.
+    has_to_be_added_to_genesis: bool,
+
+    /// Wether to register this parachain (via genesis or extrinsic).
+    /// [TODO]: is the "via genesis" part of the comment needed given the above option add to genesis ?
+    has_to_be_registered: bool,
```

Comment:
I think we can merge this options and use an enum. The possible options are:
- Register the para in the genesis
- Register the para using and extrinsic
- Not register the para.